test/test_datasets.py

```diff
@@ -141,76 +145,68 @@ def test_places365_repr_smoke(self):
-            with self.mocked_dataset(download=False):
-                pass
-
-    def test_splits(self):
-        for split in ('train', 'train+unlabeled', 'unlabeled', 'test'):
-            with self.mocked_dataset(split=split) as (dataset, data):
-                num_images = sum([data["num_images_in_split"][part] for part in split.split("+")])
-                self.generic_classification_dataset_test(dataset, num_images=num_images)
-
-    def test_folds(self):
-        for fold in range(10):
-            with self.mocked_dataset(split="train", folds=fold) as (dataset, data):
-                num_images = data["num_images_in_folds"][fold]
-                self.assertEqual(len(dataset), num_images)
-
-    def test_invalid_folds1(self):
-        with self.assertRaises(ValueError):
-            with self.mocked_dataset(folds=10):
-                pass
-
-    def test_invalid_folds2(self):
-        with self.assertRaises(ValueError):
-            with self.mocked_dataset(folds="0"):
-                pass
-
-    def test_transforms(self):
-        expected_image = "image"
-        expected_target = "target"
+    def inject_fake_data(self, tmpdir, config):
+        def make_binary_file(num_elements, root, name):
+            file_name = os.path.join(root, name)
+            np.zeros(num_elements, dtype=np.uint8).tofile(file_name)
 
-        def transform(image):
-            return expected_image
+        def make_image_file(num_images, root, name, num_channels=3, height=96, width=96):
+            make_binary_file(num_images * num_channels * height * width, root, name)
 
-        def target_transform(target):
-            return expected_target
+        def make_label_file(num_images, root, name):
+            make_binary_file(num_images, root, name)
 
-        with self.mocked_dataset(transform=transform, target_transform=target_transform) as (dataset, _):
-            actual_image, actual_target = dataset[0]
+        def make_class_names_file(root, name="class_names.txt"):
+            class_names = ("airplane", "bird")
+            with open(os.path.join(root, name), "w") as fh:
+                for cname in class_names:
+                    fh.write(f"{cname}\n")
 
-            self.assertEqual(actual_image, expected_image)
-            self.assertEqual(actual_target, expected_target)
+        def make_fold_indices_file(root):
+            num_folds = 10
+            offset = 0
+            with open(os.path.join(root, "fold_indices.txt"), "w") as fh:
+                for fold in range(num_folds):
+                    line = " ".join([str(idx) for idx in range(offset, offset + fold + 1)])
+                    fh.write(f"{line}\n")
+                    offset += fold + 1
 
-    def test_unlabeled(self):
-        with self.mocked_dataset(split="unlabeled") as (dataset, _):
-            labels = [dataset[idx][1] for idx in range(len(dataset))]
-            self.assertTrue(all([label == -1 for label in labels]))
```

Comment:
This test is missing for now. It should test that the labels in the `"unlabeled"` split always are `-1`.